### PR TITLE
Flexibus/#183/header design

### LINKED ISSUE #183: Header design
Header will be shown on the top of every page. There is no top-margin needed.
  

 
- Maximum width of content (= header from logo to login button): 1200px; Width of background (color/pic) is not limited;
- transitions when menu opens (open from top to bottom), use materialize again?
- all data (for example font size, colors etc - in mock; ask Dima for CSS);
- Font: ProximaNova

**Estimation: XL**

Version for mobile (just styling, no new menu entries, 2nd pic --> click on Dmitri --> 3rd pic)

The result should look like this (Acceptance Criteria):
 

- [ ] Implement styling of desktop view
- [ ] Implement styling of mobile view

### PR DESCRIPTION
**Pull request by emanuel-metzenthin**
Closes #183.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -48,8 +48,8 @@ GEM
       rake (>= 10.4, < 13.0)
     arel (8.0.0)
     ast (2.3.0)
-    aws-partitions (1.53.0)
-    aws-sdk-core (3.13.0)
+    aws-partitions (1.44.0)
+    aws-sdk-core (3.11.0)
```
Comment (MariusDanner) [inline Gemfile.lock]: I think downgrading is not an option here.

File: config/environments/production.rb [hunk 2]
```diff
@@ -56,8 +56,6 @@
   # Do not fallback to assets pipeline if a precompiled asset is missed.
   config.assets.compile = false
 
-  config.assets.prefix = '/dev-assets'
```
Comment (MariusDanner) [inline config/environments/production.rb]: Should definetely stay there or else the deployed version would not have the ProximaNova font

File: config/initializers/assets.rb [hunk 3]
```diff
@@ -2,7 +2,9 @@
 
 # Version of your assets, change this if you want to expire all your assets.
 Rails.application.config.assets.version = '1.0'
-Rails.application.config.assets.precompile += %w( sign_in.css navbar.css teams.css )
+Rails.application.config.assets.precompile += %w( sign_in.css )
```
Comment (MariusDanner) [inline config/initializers/assets.rb]: I think that change is unnecessary and misuses the %w() operator

File: config/routes.rb [hunk 4]
```diff
@@ -39,7 +39,6 @@
     resources :users, except: [:new, :create] do
       member do
         get 'dashboard'
-        get 'notifications'
```
Comment (MariusDanner) [inline config/routes.rb]: Why is this removed?

File: features/step_definitions/tournament.rb [hunk 5]
```diff
@@ -80,7 +80,7 @@ def find_team_of_match(match_gameday, match_num, home_or_away)
 
 Then(/^the (home|away) team of match (.+) (\d+) comes to the next round$/) do |home_or_away, match_gameday, match_num|
   team = find_team_of_match match_gameday, match_num, home_or_away
-  expect(all('a[href="' + team_path(team) + '"]').count).to eq(2)
+  all('a[href="' + team_path(team) + '"]').count == 2
```
Comment (MariusDanner) [inline features/step_definitions/tournament.rb]: Why isn't the expect function used?

File: features/step_definitions/tournament.rb [hunk 6]
```diff
@@ -100,14 +100,3 @@ def find_team_of_match(match_gameday, match_num, home_or_away)
 Given(/^a tournament with gamemode (.*)$/) do |mode|
   create_tournament game_mode: mode
 end
-
-def placing_to_display_string(placing)
-  I18n.t "events.placing.#{placing}"
-end
-
-Then(/^the (first|second) place of the tournament is the (home|away) team of (.+) (\d+)$/) do |placing, home_or_away, match_gameday, match_num|
```
Comment (MariusDanner) [inline features/step_definitions/tournament.rb]: Why do you remove this?

File: features/tournament/show_winner.feature [hunk 7]
```diff
@@ -1,14 +0,0 @@
...
-
-  Scenario: a small Tournament gets finished
-    Given a tournament t with 4 teams
-    And the Spielplan page for t is visited
-    And the results for match Halbfinale 1 (3 : 4) got inserted
-    And the results for match Halbfinale 2 (7 : 6) got inserted
-    And the results for match Finale 1 (3 : 4) got inserted
-    When the tournament's page is visited
-    Then the first place of the tournament is the home team of Halbfinale 2
-    And the second place of the tournament is the away team of Halbfinale 1
```
Comment (MariusDanner) [inline features/tournament/show_winner.feature]: Why do you remove this?

File: spec/controllers/events_controller_spec.rb [hunk 8]
```diff
@@ -344,13 +344,10 @@
...
     end
   end
-  #this test is only for leagues
+
   describe "GET League#schedule" do
     it "should generate schedule if not existing" do
       event = League.create! valid_league_attributes
-      event.add_participant(@user)
-      event.add_participant(@other_user)
-      event.generate_schedule
```
Comment (MariusDanner) [inline spec/controllers/events_controller_spec.rb]: Why do you remove this?

File: spec/routing/users_routing_spec.rb [hunk 9]
```diff
@@ -19,10 +19,6 @@
       expect(get: "/users/1/dashboard").to route_to("users#dashboard", id: "1")
     end
 
-    it "routes to #notifications" do
```
Comment (MariusDanner) [inline spec/routing/users_routing_spec.rb]: Why do you remove this?

File: spec/views/application/_navbar.html.erb_spec.rb [hunk 10]
```diff
@@ -3,6 +3,6 @@
 RSpec.describe "application/_navbar", type: :view do
   it "shows a login button" do
     render
-    expect(rendered).to have_link(t('devise.registrations.sign_in'), href: new_user_session_path)
```
Comment (MariusDanner) [inline spec/views/application/_navbar.html.erb_spec.rb:6]: Why do you remove this?

File: app/assets/stylesheets/navbar.css [hunk 11]
```diff
@@ -16,12 +94,9 @@
     border-color: transparent;
     width: 100%;
 }
-.navbar-default .navbar-brand {
-    color: white;
-}
 
 .navbar-default .navbar-nav > li > a {
-    color: white;
+    color: white !important;
```
Comment (MariusDanner) [inline app/assets/stylesheets/navbar.css:121]: I don't know if that fits here, but when checked out locally you have a white-colored font on white background
image

**Review by MariusDanner — CHANGES_REQUESTED** [review]

**Comment by MariusDanner** [comment]
It looks like all the links do not work on the mobile-friendly version
the header has the wrong background color an the mobile version too

File: app/assets/stylesheets/application.scss [hunk 12]
```diff
@@ -28,13 +28,16 @@ $secondary-color: #01A8E8 !default;
 /* Load custom font families */
 @font-face {
   font-family: 'Proxima Nova';
-  src: font_url('ProximaNova-Reg.otf') format('opentype');
+  src: url('/assets/ProximaNova-Reg.otf') format('opentype');
```
Comment (MariusDanner) [inline app/assets/stylesheets/application.scss]: That won't work. Please use the font_url()

File: app/assets/stylesheets/application.scss [hunk 13]
```diff
@@ -28,13 +28,16 @@ $secondary-color: #01A8E8 !default;
...
+  font-weight: normal;
+  font-style: normal;
 }
 
 @font-face {
-  font-family: 'Proxima Nova';
-  src: font_url('ProximaNova-Bold.otf') format('opentype');
-  font-weight: 500;
+  font-family: 'Proxima Nova Bold';
```
Comment (MariusDanner) [inline app/assets/stylesheets/application.scss]: same here

File: app/models/event.rb [hunk 14]
```diff
@@ -110,7 +110,6 @@ def standing_of(team)
   end
 
   # this is a method that simplifies manual testing, not intended for production use
-  # method not used at the moment since it is now testet with joined users
```
Comment (MariusDanner) [inline app/models/event.rb]: I don't know why you do that here

File: app/views/devise/registrations/notifications.html.erb [hunk 15]
```diff
@@ -1,4 +0,0 @@
-<%- model_class = User -%>
```
Comment (MariusDanner) [inline app/views/devise/registrations/notifications.html.erb]: Please don't remove this file

File: app/views/events/show.html.erb [hunk 16]
```diff
@@ -90,7 +89,6 @@
     <% @event.teams.each do |participant| %>
       <tr><td><%= link_to_participant participant %></td></tr>
     <% end %>
-    </tbody>
```
Comment (MariusDanner) [inline app/views/events/show.html.erb]: I think we need that too

**Review by MariusDanner — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/navbar.css (hunk 11 above)
Comment (emanuel-metzenthin) [reply app/assets/stylesheets/navbar.css:121]: Works for me

File: app/models/event.rb (hunk 14 above)
Comment (emanuel-metzenthin) [reply app/models/event.rb]: Somehow the merge went terribly wrong :)

File: app/views/application/_navbar.html.erb [hunk 17]
```diff
@@ -1,30 +1,58 @@
...
-      <%= menu_divider %>
-    <% end %>
-    <%= menu_group pull: :right do %>
-      <% if user_signed_in? %>
-        <%= drop_down (image_tag current_user.avatar_url || image_path('missing_avatar.png'), class: 'avatar') + " #{current_user.first_name}" do %>
-          <!--Fill in more drop down menu items-->
-          <%= menu_item t('devise.registrations.my_account'), edit_user_path(current_user) %>
-          <%= menu_item t('navbar.drop-down.dashboard'), dashboard_user_path(current_user) %>
-          <%= menu_item t('navbar.drop-down.edit_profile'), user_profile_edit_path(current_user) %>
-          <%= menu_item t('navbar.drop-down.notification_settings'), notifications_user_path(current_user) %>
```
Comment (MariusDanner) [inline app/views/application/_navbar.html.erb]: That needs to be there. So please don't remove it

File: spec/views/application/_navbar.html.erb_spec.rb (hunk 10 above)
Comment (Blubl) [reply spec/views/application/_navbar.html.erb_spec.rb:6]: needs to be removed because ther is now an button with an image and no text-link anymore

**Review by MariusDanner — APPROVED** [review]
Finally! Good work mateys

**Comment by stursberg** [comment]
header sollte oben bleiben -> statisch
transparent wenn banckground
events erstellen weg im header
header springt bei verschiedener breite
alles zentrieren
im mobile größer damit tippbar
weißer rand weg
